crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -276,6 +283,10 @@ pub unsafe trait ReadOnlyQueryData: QueryData<ReadOnly = Self> {}
 pub type QueryItem<'w, Q> = <Q as WorldQuery>::Item<'w>;
 /// The read-only variant of the item type returned when a [`QueryData`] is iterated over immutably
 pub type ROQueryItem<'w, D> = QueryItem<'w, <D as QueryData>::ReadOnly>;
+// TODO: Better docs, not sure about using the term "reffed".
```

Comment:
I think we can probably use something like "smart-pointer" or "change-detection enabled read-only reference".